catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessorTest.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.codice.usng4j.CoordinateSystemTranslator;
+import org.codice.usng4j.DecimalDegreesCoordinate;
+import org.codice.usng4j.UtmUpsCoordinate;
+import org.codice.usng4j.impl.CoordinateSystemTranslatorImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Matchers;
+import org.mockito.invocation.InvocationOnMock;
+
+public class UtmUpsCoordinateProcessorTest {
+  private final CoordinateSystemTranslatorImpl translatorImpl =
+      new CoordinateSystemTranslatorImpl();
+
+  private CoordinateSystemTranslator translatorMock;
+
+  private UtmUpsCoordinateProcessor processor;
+
+  @Before
+  public void before() throws ParseException {
+    translatorMock = mock(CoordinateSystemTranslator.class);
+    processor = new UtmUpsCoordinateProcessor(translatorMock);
+    when(translatorMock.parseUtmUpsString(Matchers.anyString())).thenAnswer(this::answerParse);
+    when(translatorMock.toLatLon((UtmUpsCoordinate) Matchers.anyObject()))
+        .thenAnswer(this::answerConvert);
+  }
+
+  private UtmUpsCoordinate answerParse(InvocationOnMock invocation) throws Exception {
+    return translatorImpl.parseUtmUpsString(invocation.getArgumentAt(0, String.class));
+  }
+
+  private DecimalDegreesCoordinate answerConvert(InvocationOnMock invocation) {
+    return translatorImpl.toLatLon(invocation.getArgumentAt(0, UtmUpsCoordinate.class));
+  }
+
+  @Test
+  public void testUtmStringNorthernHemisphere() {
+    processor.enhanceResults(new LinkedList<>(), "13N 234789mE 234789mN");
+    verifyInputs(1, "13N 234789mE 234789mN");
```

Comment:
❓Rather than verifying the inputs to the translator, wouldn't it be better to verify the `Suggestion` object added to the results list?

```
List<Suggestion> results = new LinkedList<>();
processor.enhanceResults(results, "...");
assertThat(results, hasSize(1));
Suggestion suggestion = results.get(0);
// verify the Suggestion
```

This way you can test the processor through only its public API and avoid using a partial mock. Moreover, you don't really care about what methods the processor calls internally to create the suggestions, just that the suggestions are correct, right?

Reply:
The issue is that I'd need to verify the actual USNG math, which is out of scope for my class's concern. I don't have the original UTM or UPS string on the suggestion object. I thought about adding it as a transient field but that seems kludgy since the UI doesn't need that information.

Granted I could assert against the complete name/label with all coordinates in square brackets, but if USNG alters its `toString()` impls my tests will fail.

Reply:
Ended up mocking out the outputs so we don't need to call into USNG just for passthrough data, per @samuelechu's suggestion below.